clientbuffer.cpp

```diff
@@ -556,13 +581,14 @@ void CClientBufferMod::UpdateTimestamp(const CClient* client, const CString& tar
 }
 #endif
 
-bool CClientBufferMod::WithinTimeLimit(const timeval& tv)
+bool CClientBufferMod::WithinTimeLimit(const timeval& tv, const CString& identifier)
 {
-	if (!m_iTimeLimit)
+    int timeLimit = GetNV(identifier + "/timelimit").ToInt();
```

Comment:
Is there a way to achieve this without introducing a `GetNV` (lookup of string key in hash map) for every operation, even when the user never used the feature added here?

Reply:
The other things are easy fixed, but this one gives me headaches :)
Can't think for now of a solution that wouldn't require storing the client timelimit somewhere and retrieving it by the identifier key. 
Do you have any thoughts on that?

Reply:
You could create some kind of an object representing the client in the OnClientLogin Method.
But since you can have many clients connecting simultaneously you would than still have to find that object referring to the given client identifier. 
So the question here is if this would be much more efficient than using the std::map stored in memory that is used as the registry?

Reply:
I think your analysis is correct. The only improvement over this that I can see would be to use the `Client*` as a map key directly, which would save us from hashing the string every time.

Let's leave this alone for now. We can revisit it later if needed.